libvast/src/system/index.cpp

```diff
@@ -688,7 +688,7 @@ caf::behavior index(caf::stateful_actor<index_state>* self, filesystem_type fs,
             self->send(client, atom::done_v);
             return;
           }
-          auto& [query_id, query_state] = *iter;
+          auto& query_state = iter->second;
```

Comment:
Independent of the syntactic change: @lava isn't capturing `iter` in the lambda a potential bug, since it may get invalidated until the lambda is called?

Reply:
Sorry, my bad. The parent context also contains an `iter`, but the local one shadows it.